### PR TITLE
[DRAFT] [UNR-3545] Experimental light cooking schema generation

### PR DESCRIPTION
**Pull request by ImprobableNic**
#### Description
Change the schema generation commandlet to use cooking services to load the assets needed to generate schema.
Need engine PR : https://github.com/improbableio/UnrealEngine/pull/518

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp [hunk 1]
```diff
@@ -161,3 +177,234 @@ int32 UCookAndGenerateSchemaCommandlet::Main(const FString& CmdLineParams)
 
 	return CookResult;
 }
+
+int32 UCookAndGenerateSchemaCommandlet::MainCook(const FString& CmdLineParams, FObjectListener& Listener)
+{
+	if (0)
```
Comment (mattyoung-improbable) [inline SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp:183]: ?

File: SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp [hunk 2]
```diff
@@ -161,3 +177,234 @@ int32 UCookAndGenerateSchemaCommandlet::Main(const FString& CmdLineParams)
...
+	{
+		return Super::Main(CmdLineParams);
+	}
+	else
+	{
+		// Copied from UCookCommandlet::CookByTheBook to fill cook parameters
```
Comment (mattyoung-improbable) [inline SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp:189]: Is all the code below copied? or are there some differences?

Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp:189]: I just skip some parameters, but no difference. We could avoid copying it if we refactored the engine cook command.

File: SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp (hunk 1 above)
Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp:183]: Just some test code. Regular cook vs minimal.

File: SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp (hunk 2 above)
Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.cpp:189]: The code I added starts line 345

**Comment by oblm** [comment]
@ImprobableNic this can be closed as a duplicate of https://github.com/spatialos/UnrealGDK/pull/2260 right?

**Pull request closed without merging**